mlf_core/create/templates/common_mlflow_files/{{ cookiecutter.common_mlflow }}/{{ cookiecutter.project_slug_no_hyphen }}/mlf_core/mlf_core.py

```diff
@@ -89,3 +90,21 @@ def log_sys_intel_conda_env(cls):
         reports_output_dir = tempfile.mkdtemp()
         cls.log_system_intelligence(reports_output_dir)
         cls.log_conda_environment(reports_output_dir)
+
+    @staticmethod
+    def md5_sum_input_data(fname: str):
+        """Generate md5 sum for input file"""
+        hash_md5 = hashlib.md5()
+        with open(fname, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+
+        md5sum = hash_md5.hexdigest()
+        return md5sum
+
+    @staticmethod
+    def log_input_data(input_data: str):
+        print('[bold blue]Hashing input data...')
+        input_hash = md5_sum_input_data(input_data)
+        print('[bold blue]Uploading input hash as a run param...')
+        mlflow.log_param("input_hash", input_hash + "-" + input_data)
```

Comment:
I think you could probably omit one of those print statements :) 

But otherwise looks nice and simple to me!

Reply:
Maybe it would be faster to just get the md5sum of the directory? Do you think we should log every input file or write it to a file for them and then log the file?

Reply:
People may not input a directory as input parameter, but files directly.
But what if we offered both options? At the moment we are not forcing them in the template as implemented anyways.

Reply:
I agree with giving the option of both. Might cut out the middle man and just do it recursively anyways though.